bin/ch/WScriptJsrt.cpp

```diff
@@ -265,6 +275,66 @@ JsValueRef WScriptJsrt::LoadScriptHelper(JsValueRef callee, bool isConstructCall
     return returnValue;
 }
 
+JsErrorCode WScriptJsrt::InitializeModuleInfo(JsValueRef specifier, JsModuleRecord moduleRecord)
+{
+    JsErrorCode errorCode = JsNoError;
+    errorCode = ChakraRTInterface::JsSetModuleHostInfo(moduleRecord, JsModuleHostInfo_FetchImportedModuleCallback, WScriptJsrt::FetchImportedModule);
+    if (errorCode == JsNoError)
+    {
+        errorCode = ChakraRTInterface::JsSetModuleHostInfo(moduleRecord, JsModuleHostInfo_NotifyModuleReadyCallback, WScriptJsrt::NotifyModuleReadyCallback);
+    }
+    if (errorCode == JsNoError)
+    {
+        errorCode = ChakraRTInterface::JsSetModuleHostInfo(moduleRecord, JsModuleHostInfo_HostDefined, specifier);
+    }
+    return errorCode;
+}
+
+JsErrorCode WScriptJsrt::LoadModuleFromString(LPCSTR fileName, LPCSTR fileContent)
+{
+    DWORD_PTR dwSourceCookie = WScriptJsrt::GetNextSourceContext();
+    JsModuleRecord requestModule = JS_INVALID_REFERENCE;
+    auto moduleRecordEntry = moduleRecordMap.find(std::string(fileName));
+    JsErrorCode errorCode = JsNoError;
+    // we need to create a new moduleRecord if the specifier (fileName) is not found; otherwise we'll use the old one.
+    if (moduleRecordEntry == moduleRecordMap.end())
+    {
+        JsValueRef specifier;
+        errorCode = ChakraRTInterface::JsPointerToStringUtf8(fileName, strlen(fileName), &specifier);
+        if (errorCode == JsNoError)
+        {
+            errorCode = ChakraRTInterface::JsInitializeModuleRecord(nullptr, specifier, &requestModule);
+        }
+        if (errorCode == JsNoError)
+        {
+            errorCode = InitializeModuleInfo(specifier, requestModule);
+        }
+        if (errorCode == JsNoError)
+        {
+            moduleRecordMap[std::string(fileName)] = requestModule;
+        }
+    }
+    else
+    {
+        requestModule = moduleRecordEntry->second;
+    }
+    if (errorCode != JsNoError)
+    {
+        return errorCode;
```

Comment:
lf this is not expected to fail, we should log the error. As the ch runs part of the unittests it will help in diagnose quickly.